augur/utils.py

```diff
@@ -83,6 +84,12 @@ def get_numerical_dates(meta_dict, name_col = None, date_col='date', fmt=None, m
                 print("WARNING: %s has an invalid data string:"%k,v)
                 continue
             elif 'XX' in v:
+                if (exclude_ambiguous_dates == 'all') or \
+                        (exclude_ambiguous_dates == 'days' and 'XX' in v.split('-')[2]) or \
+                        (exclude_ambiguous_dates == 'months' and 'XX' in v.split('-')[1]):
```

Comment:
This logic looks good. I would move this into its own simple function that takes a date string and type of date ambiguity to test and returns True/False based on the tests in this if statement. For example:

```
def is_date_ambiguous(date, ambiguous_by="all"):
    """
    Returns whether a given date string in the format of YYYY-MM-DD is ambiguous by a given part of the date (e.g., day, month, year, or all parts).

    Parameters
    ----------
    date : str
        Date string in the format of YYYY-MM-DD
    ambiguous_by : str
        Field of the date string to test for ambiguity ("day", "month", "year", "all")
    """
    year, month, day = date.split('-')
    return (
        (ambiguous_by == 'all' and 'X' in date) or
        (ambiguous_by == 'day' and 'X' in day) or
        (ambiguous_by == 'month' and 'X' in month) or
        (ambiguous_by == 'year' and 'X' in year)
    )
```

Writing out this function, it seemed a little clearer to say that a date is ambiguous by "day" than by "days" since there is only one date being considered at a time. Using "X" instead of "XX" will catch any odd cases where a date has a partial ambiguity like "200X-10-01" or "2010-10-0X".

You should be able to repurpose your unit tests below to work with this new function.